src/Instrumentation/Symfony/Messenger/EventSubscriber/EndSpanEventSubscriber.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace FriendsOfOpenTelemetry\OpenTelemetryBundle\Instrumentation\Symfony\Messenger\EventSubscriber;
+
+use FriendsOfOpenTelemetry\OpenTelemetryBundle\Instrumentation\InstrumentationTypeEnum;
+use OpenTelemetry\API\Trace\StatusCode;
+use OpenTelemetry\Context\Context;
+use OpenTelemetry\SDK\Trace\Span;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Messenger\Event\WorkerMessageFailedEvent;
+use Symfony\Component\Messenger\Event\WorkerRunningEvent;
+use Symfony\Component\Messenger\Event\WorkerStoppedEvent;
+
+class EndSpanEventSubscriber implements EventSubscriberInterface
+{
+    private ?InstrumentationTypeEnum $instrumentationType = null;
+
+    public function __construct(
+        private readonly LoggerInterface $logger
+    ) {
+
+    }
+
+    public function setInstrumentationType(InstrumentationTypeEnum $instrumentationType): void
+    {
+        $this->instrumentationType = $instrumentationType;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [
+            WorkerRunningEvent::class => ['endSpan'],
+            WorkerStoppedEvent::class => ['endSpan'],
```

Comment:
We should not end a span on running/stopped worker events but on WorkerMessageReceived and WorkerMessageHandled events.

Reply:
ended on WorkerMessageHandledEvent. But I don't understand why it should also be ended on WorkerMessageReceived

Reply:
I am sorry I mean, start a span on WorkerMessageReceived event and end span on WorkerMessageHandled event.